apps/rush-lib/src/cli/taskRunner/ProjectTask.ts

```diff
@@ -163,29 +158,30 @@ export class ProjectTask implements ITaskDefinition {
           this._hasWarningOrError = true;
         });
 
-        task.on('close', (code: number) => {
-          // Write the logs to disk
-          this._writeLogsToDisk(writer);
-
-          if (code) {
-            reject([new TaskError('error', `Returned error code: ${code}`)]);
-          } else if (this._hasWarningOrError) {
-            resolve(TaskStatus.SuccessWithWarning);
-          } else {
-            // Write deps on success.
-            if (currentPackageDeps) {
-              JsonFile.save(currentPackageDeps, currentDepsPath);
+        return new Promise((resolve: (status: TaskStatus) => void, reject: (errors: TaskError[]) => void) => {
+          task.on('close', (code: number) => {
+            // Write the logs to disk
+            this._writeLogsToDisk(writer);
+
+            if (code) {
+              reject([new TaskError('error', `Returned error code: ${code}`)]);
+            } else if (this._hasWarningOrError) {
+              resolve(TaskStatus.SuccessWithWarning);
+            } else {
+              // Write deps on success.
+              if (currentPackageDeps) {
+                JsonFile.save(currentPackageDeps, currentDepsPath);
+              }
+              resolve(TaskStatus.Success);
             }
-            resolve(TaskStatus.Success);
-          }
+          });
         });
       }
     } catch (error) {
       console.log(error);
 
-      // Write the logs to disk
       this._writeLogsToDisk(writer);
-      reject([new TaskError('error', error.toString())]);
+      return Promise.reject([new TaskError('error', error.toString())]);
```

Comment:
>[ [](start = 28, length = 1)

Why are we rejecting with an array of errors?  Errors should always be instances of the Error base class

Reply:
That's what the existing type was.

---
In reply to: [185678800](https://github.com/Microsoft/web-build-tools/pull/636#discussion_r185678800) [](ancestors = 185678800)